### PR TITLE
Sign up via GitHub

### PR DESCRIPTION
**Pull request by skddc**
As discussed, only adding routes and UI for the signup flow.

Also contains a drive-by fix for a small bug with the account name, after extracting it to a component. See commits for more info.

### CODE REVIEW

**Comment by galfert** [comment]
Wired everything up to work with 67P/hubot-kredits#41.

I think the only thing missing is showing a success message or transitioning to a success page after the contributor has been created (see https://github.com/67P/kredits-web/pull/142/files#diff-9f12217253bedb83186b9295ac7b61d2R46).

File: config/environment.js [hunk 1]
```diff
@@ -37,6 +37,9 @@ module.exports = function(environment) {
 
     web3ProviderUrl: 'https://rinkeby.infura.io/v3/d4f788b7a6584f7db2fc3c268d4d09e9',
 
+    githubConnectUrl: 'https://hal8000.chat.kosmos.org:8082/kredits/signup/connect/github',
```
Comment (skddc) [inline config/environment.js]: Why is this using a port number? This should actually fail, as the HTTPS nginx sites are not listening on those ports.

https://gitea.kosmos.org/kosmos/chef/src/branch/master/site-cookbooks/kosmos-hubot/templates/default/nginx_conf_hubot.erb#L10

Comment (galfert) [reply config/environment.js]: Oh, yes. I didn't know the actual URL, so I searched the chef repo and found this: https://gitea.kosmos.org/kosmos/chef/src/branch/master/site-cookbooks/kosmos-hubot/attributes/default.rb#L6-L7

It's fixed now.

**Comment by skddc** [comment]
As long as I haven't missed anything, I think this is good to merge.

Anyone wants to confirm?

**Comment by galfert** [comment]
Yes, I think all is good.

**Comment by skddc** [comment]
Where does the sudden merge conflict come from? :thinking:

**Comment by galfert** [comment]
No idea. Haven't edited any CSS in a long time.

**Pull request merged**